lld/ELF/MarkLive.cpp

```diff
@@ -101,6 +116,12 @@ void MarkLive<ELFT>::resolveReloc(InputSectionBase &sec, RelTy &rel,
   Symbol &sym = sec.file->getRelocTargetSym(rel);
   sym.used = true;
 
+  LiveReason reason;
+  if (!ctx.arg.whyLive.empty()) {
+    Defined *reasonSym = sec.getEnclosingSymbol(rel.r_offset);
```

Comment:
getEnclosingSymbol iterates all symbols in the file, which is very slow. This perhaps should only be run after we get a patch from a root to the --why-live specified symbol.